Build error for sanity test

I’m trying to build and install GDRCopy from source on RHEL when I had this following issue:

It seems error occurred when trying to build sanity test. Can someone please tell me how to resolve this? Looks like missing some gcc flags / gcc version here?

```
g++ -O2 -I /sw/summit/cuda/9.2.148/include -I ../include -I ../src -I /sw/summit/cuda/9.2.148/include -I /gpfs/alpine/proj-shared/csc143/jwang/local-build/check/check-0.13.0/install/include  -c -o sanity.o sanity.cpp
In file included from sanity.cpp:46:0:
common.hpp:98:9: error: 'testname' does not name a type
         testname = #__testname;                                                 \
         ^
sanity.cpp:174:1: note: in expansion of macro 'BEGIN_GDRCOPY_TEST'
 BEGIN_GDRCOPY_TEST(basic)
 ^
common.hpp:99:18: error: expected constructor, destructor, or type conversion before '(' token
         print_dbg("&&&& RUNNING " # __testname "\n");                           \
                  ^
sanity.cpp:174:1: note: in expansion of macro 'BEGIN_GDRCOPY_TEST'
 BEGIN_GDRCOPY_TEST(basic)
 ^
common.hpp:100:15: error: expected constructor, destructor, or type conversion before '(' token
         fflush(stdout);                                                         \
               ^
sanity.cpp:174:1: note: in expansion of macro 'BEGIN_GDRCOPY_TEST'
 BEGIN_GDRCOPY_TEST(basic)
 ^
common.hpp:101:15: error: expected constructor, destructor, or type conversion before '(' token
         fflush(stderr);                                                         \
               ^
sanity.cpp:174:1: note: in expansion of macro 'BEGIN_GDRCOPY_TEST'
 BEGIN_GDRCOPY_TEST(basic)
 ^
common.hpp:103:9: error: expected unqualified-id before 'if'
         if (__pid < 0) {                                                        \
         ^
sanity.cpp:174:1: note: in expansion of macro 'BEGIN_GDRCOPY_TEST'
 BEGIN_GDRCOPY_TEST(basic)
 ^
common.hpp:108:9: error: expected unqualified-id before 'if'
         if (__pid == 0)
         ^
sanity.cpp:174:1: note: in expansion of macro 'BEGIN_GDRCOPY_TEST'
 BEGIN_GDRCOPY_TEST(basic)
 ^
make[1]: *** [sanity.o] Error 1
make[1]: Leaving directory `/gpfs/alpine/csc143/proj-shared/jwang/local-build/gdrcopy/tests'
make: *** [exes] Error 2
```

System configuration:
Operating System: Red Hat Enterprise Linux Server 7.6 (Maipo)
CPU: IBM POWER9
GPU: NVIDIA Tesla v100
CUDA: 9.2.148
GCC: version 4.8.5
Check: 0.13.0
Subunit: 1.3.0

--- a/tests/common.hpp
+++ b/tests/common.hpp
@@ -94,35 +94,35 @@ namespace gdrcopy {
         extern const char *testname;
 
         #define BEGIN_GDRCOPY_TEST(__testname)                                  \
-        START_TEST(__testname)                                                  \
-        testname = #__testname;                                                 \
-        print_dbg("&&&& RUNNING " # __testname "\n");                           \
-        fflush(stdout);                                                         \
-        fflush(stderr);                                                         \
-        pid_t __pid = fork();                                                   \
-        if (__pid < 0) {                                                        \
-            print_dbg("Cannot fork\n");                                         \
-            print_dbg("&&&& FAILED " # __testname "\n");                        \
-            exit(EXIT_FAILURE);                                                 \
-        }                                                                       \
-        if (__pid == 0)
-
-        #define END_GDRCOPY_TEST                                                \
-        if (__pid) {                                                            \
-            int __child_exit_status = -EINVAL;                                  \
-            if (waitpid(__pid, &__child_exit_status, 0) == -1) {                \
-                print_dbg("waitpid returned an error\n");                       \
-                print_dbg("&&&& FAILED %s\n", gdrcopy::test::testname);         \
+        START_TEST(__testname) {                                                \
+            testname = #__testname;                                             \
+            print_dbg("&&&& RUNNING " # __testname "\n");                       \
+            fflush(stdout);                                                     \
+            fflush(stderr);                                                     \
+            pid_t __pid = fork();                                               \
+            if (__pid < 0) {                                                    \
+                print_dbg("Cannot fork\n");                                     \
+                print_dbg("&&&& FAILED " # __testname "\n");                    \
                 exit(EXIT_FAILURE);                                             \
             }                                                                   \
-            if (__child_exit_status == EXIT_SUCCESS)                            \
-                print_dbg("&&&& PASSED %s\n", gdrcopy::test::testname);         \
-            else if (__child_exit_status == EXIT_WAIVED)                        \
-                print_dbg("&&&& WAIVED %s\n", gdrcopy::test::testname);         \
-            else                                                                \
-                print_dbg("&&&& FAILED %s\n", gdrcopy::test::testname);         \
-        }                                                                       \
-        END_TEST
+            if (__pid == 0) {
+
+        #define END_GDRCOPY_TEST }                                              \
+            if (__pid) {                                                        \
+                int __child_exit_status = -EINVAL;                              \
+                if (waitpid(__pid, &__child_exit_status, 0) == -1) {            \
+                    print_dbg("waitpid returned an error\n");                   \
+                    print_dbg("&&&& FAILED %s\n", gdrcopy::test::testname);     \
+                    exit(EXIT_FAILURE);                                         \
+                }                                                               \
+                if (__child_exit_status == EXIT_SUCCESS)                        \
+                    print_dbg("&&&& PASSED %s\n", gdrcopy::test::testname);     \
+                else if (__child_exit_status == EXIT_WAIVED)                    \
+                    print_dbg("&&&& WAIVED %s\n", gdrcopy::test::testname);     \
+                else                                                            \
+                    print_dbg("&&&& FAILED %s\n", gdrcopy::test::testname);     \
+            }                                                                   \
+        } END_TEST
 
         #define ASSERT(x)                                                       \
             do                                                                  \

